src/frontends/paddle/src/input_model.cpp

```diff
@@ -201,7 +201,18 @@ std::basic_string<T> get_model_path(const std::basic_string<T>& path, std::ifstr
         // Don't throw error if file isn't opened
         // It may mean that model don't have constants
     } else {
-        model_file += paddle::get_path_sep<T>() + "__model__";
+        std::string model_name = model_file.substr(model_file.find_last_of('/') + 1, model_file.size());
```

Comment:
Don't get model path like this. The original method is ok.

Reply:
> Don't get model path like this. The original method is ok.

The original method is not suitable for the case of the model without pdiparam.

Reply:
So, do not fix like this. It adds a hidden rules that model_dir + model_name + ".pdmodel" is the model file. Let's make it clear that if the model path is a file and ends with 'pdmodel' it is model and if the model path is dir, the dir + '__model__' is the model.

Reply:
> So, do not fix like this. It adds a hidden rules that model_dir + model_name + ".pdmodel" is the model file. Let's make it clear that if the model path is a file and ends with 'pdmodel' it is model and if the model path is dir, the dir + __ model __ is the model.

Do you know where initialize the input 'path' of the InputModel?

Reply:
> So, do not fix like this. It adds a hidden rules that model_dir + model_name + ".pdmodel" is the model file. Let's make it clear that if the model path is a file and ends with 'pdmodel' it is model and if the model path is dir, the dir + __ model __ is the model.

I will change the related logic into load_impl.

Reply:
I think the logic can be change to that if the model path is a file and ends with 'pdmodel' it is model and if the model path is dir, the dir + model_name + '.pdmodel is the model. It's widely used in paddle 2.x.

Reply:
I think it's better to keep *__model__*.
Besides, I think your original fix doesn't work. What if there is a pdmodel without pdiparam and the pdmodel is not in the directory with the same name of pdmodel. In paddle 2.x, the model save rule is model_dir + model_name + ".pdmodel, but the model can be moved to other places.

Reply:
> I think it's better to keep _**model**_. Besides, I think your original fix doesn't work. What if there is a pdmodel without pdiparam and the pdmodel is not in the directory with the same name of pdmodel. In paddle 2.x, the model save rule is model_dir + model_name + ".pdmodel, but the model can be moved to other places.

You're right. I found that it's no need to change input_model.cpp. How about change `openvino/src/frontends/paddle/tests/op_fuzzy.cpp`. For these tests without model weight we can add model_name.pdmodel. For example:

```
std::string("where_index_1")  => std::string("where_index_1/where_index_1.pdmodel")
```

Reply:
No, think about the actual use case, openvino will still report error if user run a pdmodel without pdiparam

Reply:
> No, think about the actual use case, openvino will still report error if user run a pdmodel without pdiparam

It seems that we can only run a pdmodel.

<img width="1415" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/55493212/2a6a8b07-7466-4618-83d8-34d914d074da">

Actually, there were already some test case only take the pdmodel as input.

<img width="1061" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/55493212/b5f85ad2-3314-4b47-93da-5139d6454a17">

Reply:
That doesn't mean the pdmodel without pdiparam is supported.

Reply:
> That doesn't mean the pdmodel without pdiparam is supported.

How do we solve this problem? Paddle 2.6 will not generate the pdiparam file if the model don't have weights.

I tried to change op_fuzzy.cpp as bellow:

```
std::string("where_index_1")  => std::string("where_index_1/where_index_1.pdmodel")
```

After changing, this test can pass.

Reply:
> > That doesn't mean the pdmodel without pdiparam is supported.
> 
> How do we solve this problem? Paddle 2.6 will not generate the pdiparam file if the model don't have weights.
> 
> I tried to change op_fuzzy.cpp as bellow:
> 
> ```
> std::string("where_index_1")  => std::string("where_index_1/where_index_1.pdmodel")
> ```
> 
> After changing, this test can pass.

@meiyang-intel  Could you give an idea to solve this problem please? I have no idea how to deal with it.

Reply:
@meiyang-intel Hi, I found that I have a misjudgement. Instead of generating an empty pdiparam file, the old saveModel API will generate a `__model__` file, which means that previous single tests have loaded `__model__`.
".pdmodel" is not used in the previous single tests, if input path is a folder.

For the previous discussion:

"""
Besides, I think your original fix doesn't work. What if there is a pdmodel without pdiparam and the pdmodel is not in the directory with the same name of pdmodel. In paddle 2.x, the model save rule is model_dir + model_name + ".pdmodel, but the model can be moved to other places.
"""

I think there are two methods to solve this problem:

1. When the path entered by the user is a folder, check whether `model_dir + model_name + ".pdmodel` or `model_dir + model_name + __model__` exist. If they do not exist, release a load error.
2. If the user wants to load a pdmodel, let users input the pdmodel file path directly instread the folder path. What we need to do is changing the input path in op_fuzzy.cpp. I found some examples using pdmodel file as the input. https://www.paddlepaddle.org.cn/support/news?action=detail&id=2681

Besides, I still think openvino can work well if user run a pdmodel without pdiparam. I have also test it by the following codes:

![image](https://github.com/openvinotoolkit/openvino/assets/55493212/a3a05389-f031-4c67-ab1f-502328f9dc02)

![image](https://github.com/openvinotoolkit/openvino/assets/55493212/251cb785-476d-4622-a961-89ecf374f4e7)

Reply:
Hi, do you still have time to review this pr? @meiyang-intel

Reply:
Sorry for the late reply. I have a concern that whether it still works if 'round.pdmodel' is not in directory 'round'

Reply:
> Sorry for the late reply. I have a concern that whether it

```
std::string model_name = model_file.substr(model_file.find_last_of('/') + 1, model_file.size());
```

I think it will have errors for this case.

Reply:
> Sorry for the late reply. I have a concern that whether it still works if 'round.pdmodel' is not in directory 'round'

For this case use should input the path of pdmodel. Openvino will found the pdiparam in the save dir. Could you see this example? 

https://www.paddlepaddle.org.cn/support/news?action=detail&id=2681

They use the following script:

```
python mo.py --model_name yolov3_mobilenet_v1_270e_coco \
--output_dir <PATH_TO_OUTPUT_DIR> \
--framework=paddle \
--data_type=FP32 \
--reverse_input_channels \
--input_shape=[2,3,608,608],[1,2],[1,2] \
--input=image,im_shape,scale_factor \
--output=save_infer_model/scale_0.tmp_1,save_infer_model/scale_1.tmp_1 \
--input_model= < PATH_TO_MODEL_DIR\model.pdmodel>
```

`--input_model` directly gives the path of the input model.

@meiyang-intel

Reply:
Of course it can find the pdiparam, it is how OpenVINO read the pdmodel and pdimparam. And there is no rule to force model.pdmodel have to be put in the directory of 'model'.
But the problem is OpenVINO will report error if there is no pdiparams. I review your code, seems you didn't solve this problem. The current code is just a work around to get the pdiparam in the directory. As I see, this work around will not work if the pdmodel is not in the directory with the same as model. Besides, this work around will also change the behavior how OpenVINO read the paddle model.

Reply:
> Of course it can find the pdiparam, it is how OpenVINO read the pdmodel and pdimparam. And there is no rule to force model.pdmodel have to be put in the directory of 'model'. But the problem is OpenVINO will report error if there is no pdiparams. I review your code, seems you didn't solve this problem. The current code is just a work around to get the pdiparam in the directory. As I see, this work around will not work if the pdmodel is not in the directory with the same as model. Besides, this work around will also change the behavior how OpenVINO read the paddle model.

@meiyang-intel  You are right. How about recover this change about reading paddle model and change the unit tests' input path from folder to the pdmodel's path?

After recover the code, there is no read error when pdiparam file doesn't exist.

I have also test input pdmodel without pdiparam by the following codes:

![image](https://github.com/openvinotoolkit/openvino/assets/55493212/e9ba93a6-1452-401d-96e5-dddbc1871eaf)

There is no pdiparams for round.pdmodel. But Openvino can read the model and pass the test.

![image](https://github.com/openvinotoolkit/openvino/assets/55493212/39ba553c-0339-4859-afc3-bee5011ea1a9)

Reply:
yes, it's ok

Reply:
> yes, it's ok

done, https://github.com/openvinotoolkit/openvino/pull/23010/commits/4627364b1bf2f8ed0c70107fe20365c00cf4109b

Could you help to run the workflows? @meiyang-intel